lgc/state/PipelineState.cpp

```diff
@@ -988,8 +988,14 @@ bool PipelineState::isTessOffChip() {
 bool PipelineState::isPackInOut() {
   // Pack input/output requirements:
   // 1) -pack-in-out option is on
-  // 2) It is a VS-FS pipeline
-  return PackInOut && (m_stageMask == (shaderStageToMask(ShaderStageVertex) | shaderStageToMask(ShaderStageFragment)));
+  // 2) It supports VS-FS, VS-TCS-TES-(FS)
+  if (!PackInOut)
+    return false;
+
+  const static unsigned vsFsMask = shaderStageToMask(ShaderStageVertex) | shaderStageToMask(ShaderStageFragment);
+  const static unsigned tessMask = shaderStageToMask(ShaderStageVertex) | shaderStageToMask(ShaderStageTessControl) |
+                                   shaderStageToMask(ShaderStageTessEval);
+  return m_stageMask == vsFsMask || m_stageMask == tessMask;
```

Comment:
Thanks, but couldn't you allow ShaderStageFragment as well in the `tessMask` case? You only really need to forbid GS, so perhaps the check could just be something like `!hasShaderStage(ShaderStageGeometry)`?

Reply:
Yes, you are right. Fixed.